### ISSUE TITLE
Product Grid Item throws TypeError if no price is defined for a product [PR provided]

### ISSUE THREAD

**Issue by ESchuderer** [SAP/cloud-commerce-spartacus-storefront#3765]
# Environment Details

- Spartacus:  1.0.1

## Steps to Reproduce
Define a product without PriceRows in SAP Commerce and search for it, then switch to the grid display.

## Observed Results

TypeError is thrown, no further products are rendered.

## Expected Results

No error is thrown, all products are rendered.

## Repository Used
-

## Offer to Fix
Yes

## Additional Information
The ProductListItem component does check the price attribute, therefore I assume this is also the intended behavior for the ProductGridItem component.

**Comment by hackergil** [comment]
Depends on #4386 to be merged

**Comment by Xymmer** [comment]
this issue is big enough that we should redo the PR so no one has to wait.
as we still aren't set up for external PRs yet.

**Comment by KateChuen** [comment]
Tested on dev10. Not sure I like the fix. 
I think an improvement should be done along with the fix, which is: disabling the "Add to cart" button. Because:
- At some point, after removing the price, I was able to add the item to cart, and the price was $0. 
![Screen Shot 2019-10-07 at 3 12 18 PM](https://user-images.githubusercontent.com/45409515/66341887-1ad17e00-e916-11e9-83ec-f49f5da5a0f2.png)

- At other time, the price in the cart was back to its original value, even though on the pdp and plp, it doesn't have any price set to it. And I'm unable to remove the item from cart, update the qty or checkout. So I'm stuck. 
![Screen Shot 2019-10-07 at 3 19 58 PM](https://user-images.githubusercontent.com/45409515/66341910-291f9a00-e916-11e9-97f3-540d9dbeea9f.png)

@Xymmer Your input on this please? Proceed with this ticket and create another ticket to improve the behavior?

**Comment by KateChuen** [comment]
Created ticket https://github.com/SAP/cloud-commerce-spartacus-storefront/issues/4877 to tackle the above issues.
